### ISSUE TITLE
VBO: Do not allow blocking the current user, or user with UID 1.

### ISSUE THREAD

**Issue by serundeputy** [backdrop/backdrop-issues#1500]
I think if a user is bulk deleting users and User 1 is selected it should not block the user 1 account.

Steps to reproduce (Backdrop 1.x):
- Loggin as user 1
- Create some users
- visit: `/admin/people/list`
- Select all the users
- Under Operations dropdown select Block User(s)
- Click execute

You are logged out and the DB shows user 1 status = 0;

<img width="1185" alt="screen shot 2016-01-09 at 1 29 43 pm" src="https://cloud.githubusercontent.com/assets/6763671/12217765/4f661bae-b6d8-11e5-9bbe-0a16115643f8.png">

It blocks all the users including you (user 1); I don't think this is good or desired behaviour.

We should add an exception for user 1.

**Comment by docwilmot** [comment]
Oh dear.

**Comment by klonos** [comment]
> We should add an exception for user 1.

+ the user that is performing the action (if it is not user 1).

**Comment by BWPanda** [comment]
I've made a PR to fix this. It prevents the current user account and the user 1 account from being blocked (displaying a message and returning FALSE instead).

In order to avoid those accounts being counted as part of the 'Block user(s) was applied to N items' message, I also had to tweak the VBO code to not count items that returned FALSE.

**Comment by olafgrabienski** [comment]
Not fully tested but works for me in a quick PR sandbox test: Tried to block User 1 using the bulk action. User 1 is still active, and I got the message "You cannot block your own user account."

**Comment by indigoxela** [comment]
Oops what a nasty bug...

This is how I tested:

- Created a privileged user "testadmin"
- Logged in as that user, went to admin/people/list
- Selected all rows
- Applied action "Block user(s)"

Messages:
```
You cannot block your own user account.
Status messageBlock user(s) was applied to 1 item.
```
Uhoh, user 1 was blocked. Guess what - `$account->uid` is a string, not an integer.

**Comment by BWPanda** [comment]
> `$account->uid` is a string, not an integer.

And *that's* why we have reviews/testing :laughing: Thanks @indigoxela! PR updated.

**Comment by indigoxela** [comment]
This time I tested in the sandbox - which already has an additional user.

- Added a privileged user "testadmin"
- Logged in as "testadmin"
- Unblocked that previously blocked user "test1"
- Selected all rows, applied action "Block user(s)

Messages:
```
You cannot block your own user account.
User 1 cannot be blocked.
Block user(s) was applied to 1 item.
```
Perfect, neither testadmin (= current user) nor user 1 have been blocked, test1 has been blocked.

**Comment by klonos** [comment]
> `$account->uid` is a string, not an integer.

We should fix that (separate issue though).

@BWPanda I've left a couple of comments re usage of single/double quotes (see https://api.backdropcms.org/php-standards#quotes) + a suggestion to include the actual account name in the warning. Other than that, this is RTBC 👍

Thanks

**Comment by BWPanda** [comment]
Thanks @klonos. PR updated.

**Comment by klonos** [comment]
Thank YOU @BWPanda ...RTBC 👍

**Comment by BWPanda** [comment]
Thanks @serundeputy for raising this issue! Thanks also to @olafgrabienski, @indigoxela & @klonos for testing. I've merged https://github.com/backdrop/backdrop/pull/3231 into 1.x and 1.16.x.
